soh/soh/Enhancements/debugger/MessageViewer.cpp

```diff
@@ -0,0 +1,278 @@
+    if (ImGui::Button("Display Message##CustomMessage")) {
+        mDisplayCustomMessageClicked = true;
+    }
+    ImGui::End();
+    // ReSharper restore CppDFAUnreachableCode
+}
+
+void MessageViewer::UpdateElement() {
+    if (mDisplayExistingMessageClicked) {
+        mTableId = std::string(mTableIdBuf);
+        switch (mTextIdBase) {
+            case DECIMAL:
+                mTextId = std::stoi(std::string(mTextIdBuf), nullptr, 10);
+                break;
+            case HEXADECIMAL:
+            default:
+                mTextId = std::stoi(std::string(mTextIdBuf), nullptr, 16);
+                break;
+        }
+        DisplayExistingMessage();
+        mDisplayExistingMessageClicked = false;
+    }
+    if (mDisplayCustomMessageClicked) {
+        mCustomMessageString = std::string(mCustomMessageBuf);
+        std::erase(mCustomMessageString, '\n');
+        DisplayCustomMessage();
+        mDisplayCustomMessageClicked = false;
+    }
+}
+
+void MessageViewer::DisplayExistingMessage() const {
+    MessageDebug_StartTextBox(mTableId.c_str(), mTextId, mLanguage);
+}
+
+void MessageViewer::DisplayCustomMessage() const {
+    MessageDebug_DisplayCustomMessage(mCustomMessageString.c_str());
+}
+
+extern "C" MessageTableEntry* sNesMessageEntryTablePtr;
+extern "C" MessageTableEntry* sGerMessageEntryTablePtr;
+extern "C" MessageTableEntry* sFraMessageEntryTablePtr;
+extern "C" MessageTableEntry* sStaffMessageEntryTablePtr;
+
+void FindMessage(PlayState* play, const uint16_t textId, const uint8_t language) {
+    const char* foundSeg;
+    const char* nextSeg;
+    MessageTableEntry* messageTableEntry = sNesMessageEntryTablePtr;
+    Font* font;
+    u16 bufferId = textId;
+    // Use the better owl message if better owl is enabled
+    if (CVarGetInteger("gBetterOwl", 0) != 0 && (bufferId == 0x2066 || bufferId == 0x607B ||
+        bufferId == 0x10C2 || bufferId == 0x10C6 || bufferId == 0x206A))
+    {
+        bufferId = 0x71B3;
+    }
+
+    if (language == LANGUAGE_GER)
+        messageTableEntry = sGerMessageEntryTablePtr;
+    else if (language == LANGUAGE_FRA)
+        messageTableEntry = sFraMessageEntryTablePtr;
+
+    // If PAL languages are not present in the OTR file, default to English
+    if (messageTableEntry == nullptr)
+        messageTableEntry = sNesMessageEntryTablePtr;
+
+    const char* seg = messageTableEntry->segment;
+
+    while (messageTableEntry->textId != 0xFFFF) {
+        font = &play->msgCtx.font;
+
+        if (messageTableEntry->textId == bufferId) {
+            foundSeg = messageTableEntry->segment;
+            font->charTexBuf[0] = messageTableEntry->typePos;
+
+            nextSeg = messageTableEntry->segment;
+            font->msgOffset = reinterpret_cast<uintptr_t>(messageTableEntry->segment);
+            font->msgLength = messageTableEntry->msgSize;
+
+            // "Message found!!!"
+            osSyncPrintf(" メッセージが,見つかった！！！ = %x  "
```

Comment:
I would remove these logs as they're not helpful imo.

Reply:
Good point will do.